ui/src/localPeer.ts

```diff
@@ -69,20 +70,29 @@ export type Event =
   | RequestTimedOut
   | { type: EventType.StatusChanged; old: Status; new: Status };
 
-// STATE
-const eventStore = writable<Event | null>(null, set => {
-  const source = new EventSource(
-    "http://localhost:8080/v1/notifications/local_peer_events"
-  );
-
-  source.addEventListener("message", (msg: MessageEvent): void => {
-    const event: Event = JSON.parse(msg.data);
-    set(event);
-  });
-
-  return (): void => source.close();
+let eventSource: EventSource | null = null;
+
+session.session.subscribe(sess => {
+  if (
+    sess.status === remote.Status.Success &&
+    sess.data.status === session.Status.UnsealedSession
+  ) {
+    if (eventSource === null || eventSource.readyState === EventSource.CLOSED) {
+      eventSource = new EventSource(
```

Comment:
This is what I tried initially to tackle the problem. It turned out that this was more involved than expected and I opted for this version. Since this is only incidentally related and not a regression I’d prefer to tackle this in a separate issue.